File: backend/app/api/agents.py
"""Agent API routes."""

import uuid

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession

from app.database.session import get_db_session
from app.schemas.schemas import AgentCreate, AgentRead, AgentUpdate
from app.services.agent_service import AgentService

router = APIRouter(prefix="/workspaces/{workspace_id}/agents", tags=["agents"])


@router.post("/", response_model=AgentRead, status_code=201)
async def create_agent(
    workspace_id: uuid.UUID,
    data: AgentCreate,
    session: AsyncSession = Depends(get_db_session),
) -> AgentRead:
    service = AgentService(session)
    return await service.create_agent(workspace_id, data)


@router.get("/", response_model=list[AgentRead])
async def list_agents(
    workspace_id: uuid.UUID,
    limit: int = 50,
    offset: int = 0,
    session: AsyncSession = Depends(get_db_session),
) -> list[AgentRead]:
    service = AgentService(session)
    return await service.list_agents(workspace_id, limit=limit, offset=offset)


@router.get("/{agent_id}", response_model=AgentRead)
async def get_agent(
    workspace_id: uuid.UUID,
    agent_id: uuid.UUID,
    session: AsyncSession = Depends(get_db_session),
) -> AgentRead:
    service = AgentService(session)
    agent = await service.get_agent(agent_id)
    if agent is None or agent.workspace_id != workspace_id:
        raise HTTPException(status_code=404, detail="Agent not found")
    return agent


@router.patch("/{agent_id}", response_model=AgentRead)
async def update_agent(
    workspace_id: uuid.UUID,
    agent_id: uuid.UUID,
    data: AgentUpdate,
    session: AsyncSession = Depends(get_db_session),
) -> AgentRead:
    service = AgentService(session)
    agent = await service.update_agent(agent_id, data)
    if agent is None:
        raise HTTPException(status_code=404, detail="Agent not found")
    return agent


@router.delete("/{agent_id}", status_code=204)
async def delete_agent(
    workspace_id: uuid.UUID,
    agent_id: uuid.UUID,
    session: AsyncSession = Depends(get_db_session),
) -> None:
    service = AgentService(session)
    deleted = await service.delete_agent(agent_id)
    if not deleted:
        raise HTTPException(status_code=404, detail="Agent not found")

File: apps/desktop/src/api/agents.ts
import { apiGet, apiPost, apiPatch, apiDelete } from './client';
import type { Agent, CreateAgent, UpdateAgent } from './types';

export const agentsApi = {
  list(workspaceId: string, limit = 50, offset = 0): Promise<Agent[]> {
    return apiGet<Agent[]>(`/workspaces/${workspaceId}/agents/?limit=${limit}&offset=${offset}`);
  },

  get(workspaceId: string, agentId: string): Promise<Agent> {
    return apiGet<Agent>(`/workspaces/${workspaceId}/agents/${agentId}`);
  },

  create(workspaceId: string, data: CreateAgent): Promise<Agent> {
    return apiPost<Agent>(`/workspaces/${workspaceId}/agents/`, data);
  },

  update(workspaceId: string, agentId: string, data: UpdateAgent): Promise<Agent> {
    return apiPatch<Agent>(`/workspaces/${workspaceId}/agents/${agentId}`, data);
  },

  delete(workspaceId: string, agentId: string): Promise<void> {
    return apiDelete<void>(`/workspaces/${workspaceId}/agents/${agentId}`);
  },
};
